amzn2023/nvidia-driver

```diff
@@ -0,0 +1,776 @@
+#! /bin/bash
+# Copyright (c) 2018-2020, NVIDIA CORPORATION. All rights reserved.
+
+set -eu
+
+RUN_DIR=/run/nvidia
+PID_FILE=${RUN_DIR}/${0##*/}.pid
+DRIVER_VERSION=${DRIVER_VERSION:?"Missing DRIVER_VERSION env"}
+KERNEL_UPDATE_HOOK=/run/kernel/postinst.d/update-nvidia-driver
+NUM_VGPU_DEVICES=0
+GPU_DIRECT_RDMA_ENABLED="${GPU_DIRECT_RDMA_ENABLED:-false}"
+USE_HOST_MOFED="${USE_HOST_MOFED:-false}"
+NVIDIA_MODULE_PARAMS=()
+NVIDIA_UVM_MODULE_PARAMS=()
+NVIDIA_MODESET_MODULE_PARAMS=()
+NVIDIA_PEERMEM_MODULE_PARAMS=()
+TARGETARCH=${TARGETARCH:?"Missing TARGETARCH env"}
+
+OPEN_KERNEL_MODULES_ENABLED=${OPEN_KERNEL_MODULES_ENABLED:-false}
+[[ "${OPEN_KERNEL_MODULES_ENABLED}" == "true" ]] && KERNEL_TYPE=kernel-open || KERNEL_TYPE=kernel
+
+DRIVER_ARCH=${TARGETARCH/amd64/x86_64} && DRIVER_ARCH=${DRIVER_ARCH/arm64/aarch64}
+
+echo "DRIVER_ARCH is $DRIVER_ARCH"
+
+_update_package_cache() {
+    if [ "${PACKAGE_TAG:-}" != "builtin" ]; then
+        echo "Updating the package cache..."
+        dnf -q makecache
+    fi
+}
+
+_cleanup_package_cache() {
+    if [ "${PACKAGE_TAG:-}" != "builtin" ]; then
+        echo "Cleaning up the package cache..."
+        yum clean all
+        dnf clean
+        rm -rf /var/cache/yum/*
+    fi
+}
+
+_update_ca_certificates() {
+    if [ ! -z "$(ls -A /etc/pki/ca-trust/source/anchors/ )" ]; then
+        update-ca-trust extract
+    fi
+}
+
+# Resolve the kernel version to the form major.minor.patch-revision-flavor where flavor defaults to generic.
+_resolve_kernel_version() {
+    dnf -y -q --releasever=latest update >/dev/null
+    dnf -y -q groupinstall "Development Tools" >/dev/null
+
+    local version=$(yum info "kernel-headers-${KERNEL_VERSION}" 2> /dev/null | \
+        grep Version | sed -nE 's/^Version\s+:\s+(([0-9]+\.){2}[0-9]+)([-.]([0-9]+))?.*/\1\4/p')
+    local flavor=$(echo ${KERNEL_VERSION} | sed 's/[^a-z]*//' | grep -Ev "^generic|virtual")
+
+    echo "Resolving Linux kernel version..."
+    if [ -z "${version}" ]; then
+        echo "Could not resolve Linux kernel version" >&2
+        return 1
+    fi
+
+    # TODO check later version, when kernel source released
+    # currently linux headers not available
+    # linux headers with extension of amzn2023 is available
+    # try to compile it with that only
+    # fake path KERNEL_VERSION_PATH added for installation
+    # KERNEL_VERSION  is not disturbed
+    # KERNEL_VERSION="${version}-${flavor:-generic}"
+    KERNEL_VERSION_PATH="${version}-${flavor:-generic}-nvidia"
```

Comment:
I am not following this... could you explain the issue you were facing here? Maybe an example kernel version string you are observing would help.

Reply:
updated the code